### PR TITLE
chore(v2): remove redundant mlmd call and fixed flag parse bug

### PR DESCRIPTION
**Pull request by capri-xiyue**
**Description of your changes:**

1. remove redundant mlmd call
2. fixed flag parse bug

**Checklist:**
- [ ] The title for your pull request (PR) should follow our title convention. [Learn more about the pull request title convention used in this repository](https://github.com/kubeflow/pipelines/blob/master/CONTRIBUTING.md#pull-request-title-convention).

### CODE REVIEW

File: v2/cmd/launch/main.go [hunk 1]
```diff
@@ -31,13 +32,18 @@ var (
...
 	pipelineRoot      = flag.String("pipeline_root", "", "The root output directory in which to store output artifacts.")
-	enableCaching     = flag.Bool(   "enable_caching", false, "Enable caching or not")
+	enableCaching     = flag.String(   "enable_caching", "false", "Enable caching or not")
 )
 
 func main() {
 	flag.Parse()
 	ctx := context.Background()
 
+	enableCachingBool, err := strconv.ParseBool(*enableCaching)
```
Comment (Bobgy) [inline v2/cmd/launch/main.go:42]: for my education, what's the difference between this and flag.Bool? maybe comment the reason here?

**Review by Bobgy — COMMENTED** [review]
thanks!
/lgtm
/approve

File: v2/cmd/launch/main.go (hunk 1 above)
Comment (capri-xiyue) [reply v2/cmd/launch/main.go:42]: When using flag.Bool directly, the result of flags.args here have `true` in the array which will break parser logic https://github.com/kubeflow/pipelines/blob/c6cb8acf7a2b3a9361b02e059249433538293c9b/v2/component/launcher.go#L229 
Screen Shot 2021-07-13 at 6 58 53 PM

Comment (Bobgy) [reply v2/cmd/launch/main.go:44]: Ohh, you are right. `flag.Bool` does not support `--enable_caching true` syntax,
see https://pkg.go.dev/flag#hdr-Command_line_flag_syntax

> -flag
-flag=x
-flag x  // non-boolean flags only

Comment (Bobgy) [reply v2/cmd/launch/main.go:44]: @capri-xiyue can you include the ref in the comments too, I think that makes it clearer.
The root cause is that `-flag x` format is not supported for boolean flag, therefore the `true` item is not recognized and left in the args array.

Comment (capri-xiyue) [reply v2/cmd/launch/main.go:44]: done

**Pull request merged**